.github/workflows/build_and_test.yml

```diff
@@ -19,6 +19,7 @@ jobs:
       - run: RUSTFLAGS="-D warnings -F unsafe-code" cargo build --verbose
       - run: cargo doc --all-features --no-deps
       - run: cargo clippy --all-features
```

Comment:
We want to add `-D` to this line, but clippy wants us to add `#[must_use]` to a bunch of functions in api.rs, so I'll do than in another PR after we discuss whether than needs a semver bump.